book/analysis/performance-of-python-types.md

```diff
@@ -5,90 +5,74 @@ position: 4
-effectively we must iterate over every element of the list to determine
-if the element in question is contained therein.
-
-The slice operations require some more thought. To access a slice `[a:b]`
-of an array, we must iterate through every element between that at index
-`a` and that at index `b`. As such slice access is $$O(k)$$ where $$k$$
-is the number of elements in the slice. Deleting a slice is $$O(n)$$ for
-the same reason that deleting a single element is—$$n$$ subsequent
-elements must then be shifted toward the beginning of the list.
-
-As mentioned previously, concatenation is $$O(k)$$ where $$k$$ is the
-length of the concatenated list, due to an effective $$k$$ appends.
-It follows that multiplying a list is $$O(nk)$$ as multiplication of
-a $$k$$ length list $$n$$ times will involve $$k(n - 1)$$ appends, so
-in the order of $$nk$$.
-
-Reversing a list in $$O(n)$$ as each element must be repositioned.
-Finally, as previously mentioned [sort in Python](http://svn.python.org/view/python/trunk/Objects/listsort.txt?revision=69846&view=markup) is $$O(n log n)$$, the
-least intuitive of all of these outcomes, and [beyond the scope of this book](https://en.wikipedia.org/wiki/Timsort)
-to demonstrate.
-
-Below we provide a table of performance characteristics of Python list
-operations, for future reference.
+Python designers had many choices to make when they implemented the list data
+type. Each choice has a potential impact on how quickly the list can perform
+operations. One decision they made was to optimize the list implementation for
+commonly-used operations. This doesn't mean rarer operations are slow; it just
+means that common operations got preference during situations where the
+designers had to make a tradeoff.
+
+Two common operations are indexing and assigning to an index position. In
+Python lists, values are assigned to and retrieved from specific, known memory
+locations. No matter how large the list is, then, index lookup and assignment
+take a constant amount of time and are thus $$O(1)$$.
+
+Another common programming need is to grow a list. There are ways to do this:
+you can use the `append` method or the concatenation operator (`+`).
+
+The `append` method is $$O(1)$$ since the new value is stored at a specific
+memory location that has already been allocated for the list, much like index assignment. However, concatenation is $$O(k)$$, where $$k$$ is the size of the
+concatenated list, since $$k$$ sequential assignment operations must occur.
+
+Popping from a Python list is typically performed from its end but, by passing
+an index, you can pop from a specific position. When `pop` is called from the
+end, the operation is $$O(1)$$, while calling `pop` from anywhere else is
+$$O(n)$$. Why the difference?
+
+When an item is taken from the front of a Python list, all other elements in
+the list are shifted one position closer to the beginning. This might seem
```

Comment:
How about instead of "This might seem silly..." just "This is an unavoidable cost of $$O(1)$$ index lookup, which is the more common operation".